src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcEndpointDataSource.cs

```diff
@@ -442,4 +429,19 @@ public RouteNameMetadata(string routeName)
             public string Name { get; }
         }
     }
+
+    internal class DummyRouter : IRouter
```

Comment:
make this a top level class, preferrably also rename to `NullRouter` for consistency with routing.